pkg/maps/cidrmap/cidrmap.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2016-2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cidrmap
+
+/*
+#cgo CFLAGS: -I../../../bpf/include
+#include <linux/bpf.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"net"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	MapName = "cilium_cidr_"
+)
+
+type CIDRMap struct {
+	path     string
+	Fd       int
+	AddrSize int // max prefix length in bytes, 4 for IPv4, 16 for IPv6
+}
+
+func (cm *CIDRMap) DeepCopy() *CIDRMap {
+	return &CIDRMap{
+		path:     cm.path,
+		Fd:       cm.Fd,
+		AddrSize: cm.AddrSize,
+	}
+}
+
+const (
+	MAX_KEYS = 1024
+)
+
+func (pe *PolicyEntry) String() string {
+	return string(pe.Action)
+}
+
+type PolicyEntry struct {
+	Action  uint32
+	Pad     uint32
+	Packets uint64
+	Bytes   uint64
+}
+
+func (pe *PolicyEntry) Add(oPe PolicyEntry) {
+	pe.Packets += oPe.Packets
+	pe.Bytes += oPe.Bytes
+}
+
+type CIDRKey struct {
+	Prefixlen uint32
+	Net       [16]byte
+}
+
+func (cm *CIDRMap) CIDRKeyInit(cidr net.IPNet) CIDRKey {
+	var key CIDRKey
+	ones, _ := cidr.Mask.Size()
+	key.Prefixlen = uint32(ones)
+	// IPv4 address can be represented by 16 byte slice in 'cidr.IP',
+	// in which case the address is at the end of the slice.
+	copy(key.Net[:], cidr.IP[len(cidr.IP)-cm.AddrSize:len(cidr.IP)])
+	return key
+}
+
+func (cm *CIDRMap) AllowCIDR(cidr net.IPNet) error {
+	key := cm.CIDRKeyInit(cidr)
+	entry := PolicyEntry{Action: 1}
```

Comment:
Currently even the policy entry counters are not being updated, as LPM maps can not even be dumped. I'll change the code to not use the policy entry at all to make this clearer.